core/node/node_storage_init/src/main_node/genesis.rs

```diff
@@ -30,6 +30,12 @@ impl InitializeStorage for MainNodeGenesis {
         }
 
         let params = GenesisParams::load_genesis_params(self.genesis.clone())?;
+        zksync_node_genesis::validate_genesis_params(
+            &params,
+            &self.l1_client,
+            self.contracts.diamond_proxy_addr,
+        )
+        .await?;
```

Comment:
This check would only be performed on a fresh setup, so if someone will change the parameters afterwards -- it won't be checked. Is that expected?

Reply:
Yes, they are inserted into db after fresh setup so it's fine if they are changed later as long as genesis config is not used (it shouldn't be changed nor used)